lib/tbot/loop.go

```diff
@@ -26,10 +26,40 @@ import (
 
 	"github.com/gravitational/trace"
 	"github.com/jonboulle/clockwork"
+	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/gravitational/teleport/api/utils/retryutils"
 )
 
+var (
+	loopIterationsCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "tbot_task_iterations",
+			Help: "Number of task iteration attempts, not counting retries",
+		}, []string{"name"},
+	)
+	loopIterationsSuccessCounter = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "tbot_task_iterations_successful",
+			Help:    "Histogram of task iterations that ultimately succeeded, bucketed by number of retries before success",
+			Buckets: []float64{0, 1, 2, 3, 4, 5},
+		}, []string{"name"},
+	)
+	loopIterationsFailureCounter = prometheus.NewCounterVec(
```

Comment:
I tend to see this grouped with `tbot_task_iterations` using some kind of label to indicate status.

Reply:
I think since we're tracking successful iterations via a histogram, the `failed` state - which would've otherwise been grouped into a labelled counter alongside `successful` - gets left out on its own. If you'd like, I could group them anyway and record `tbot_task_iterations{status="successful"}` as a duplicate of `tbot_task_iterations_successful_count`? (That one's recorded automatically as part of the histogram)

I've at least renamed `tbot_task_iterations` to `tbot_task_iterations_total` since I think that's a bit more in line with Prometheus conventions. That one definitely needs to remain separate, I think.

Reply:
I figured that the histogram could also have a "state" label - but this is fine how it is if you'd rather proceed as is.